src/app/selectors.js

```diff
@@ -372,37 +372,53 @@ const filterModelData = (filters, modelData, controllers) => {
     return clonedModelData;
   }
   const filterSegments = {};
-  // Collect segments
-  filters.forEach((filter) => {
-    const values = filter.split(":");
-    if (!filterSegments[values[0]]) {
-      filterSegments[values[0]] = [];
+
+  // Collect segments from filter data
+  Object.entries(filters).forEach((filter) => {
+    if (filter[1].length === 0) return;
+    if (!filterSegments[filter[0]]) {
+      filterSegments[filter[0]] = [];
     }
-    filterSegments[values[0]].push(values[1]);
+    filterSegments[filter[0]].push(filter[1]);
   });
 
   Object.entries(clonedModelData).forEach(([uuid, data]) => {
```

Comment:
This seems like duplicated functionality with useModelAttributes?

Reply:
`useModelAttributes` takes all available models and pull out the filters that can be applied to them based on the attributes available.. owner, region, etc

This selector takes the filters which have been added to the Search and Filter component and segments them so they can be used to filter out the model data.